course/models.py

```diff
@@ -142,7 +149,45 @@ def create_enrollment_code(sender, instance, created, **kwargs):
         instance.personal_code = code
         instance.save()
 
+def create_anon_id(sender, instance, created, **kwargs):
+    if created or not instance.anon_id:
+        nums = '0123456789'
```

Comment:
10^16 is good enough, but this could also include small letters (7.9*10^24)..

in addition. we can use constants:
```python
import string
nums= string.digits + string.ascii_lowercase
```